### ISSUE TITLE
(kind of urgent) Feature request: Add "ChromeBook Plus" to the Manifest.

### ISSUE THREAD

**Issue by noseshimself** [stefan-niedermann/nextcloud-deck#1541]
**Is your feature request related to a problem? Please describe.**
The application can't be installed on a number of ChromeOS devices. After a comparison of similar problems with other applications and other users it is apparent that the applications are marked as incompatible with "ChromeBook Plus" devices. This is consistent with the use of that marketing designation that also offers certain applications that are not running on non-"ChromeBook Plus" devices...

**Describe the solution you'd like**
Add the necessary capability to your manifests that permits installation not only on ChromeOS but also on the stub-type "Chromebook Plus" and release it on the Google Play store (the only source of Android applications for ChromeOS).
As there is another "mostly marketing and obviously failing" sub-type of ChromeBooks for gaming (permitted to install Steam) it might be possible that installation is prohibited on these machines, too (as I don't have one I can't check that).

**Describe alternatives you've considered**
Selling the ChromeBook Plus and getting a mass-market ChromeBook instead -- but all higher-end devices have become members of the "Plus" family these days.

**additional confusing hint**
About two hours ago the installation of Nextcloud Tables (which was also flagged as incompatible) started to work on the devices...

**Comment by stefan-niedermann** [comment]
I am kind of agnostic about this topic but I am willing to accept a contribution if you

1. open a Pull Request with the necessary changes, test the changes locally and post a screenshot of the result
2. ensure that the main Nextcloud Android app already runs on ChromeOS
3. you are commit to maintain this feature as we have no ChromeOS available for testing purposes

Given the Single Sign On mechanism we are relying on uses quite low level [AIDL](https://developer.android.com/develop/background-work/services/aidl) I would not bet that the necessary communication between the Deck Android app and the main Nextcloud Android app will work.

Do you agree with these conditions?

**Comment by noseshimself** [comment]
> I am kind of agnostic about this topic but I am willing to accept a contribution if you
> 
> 1. open a Pull Request with the necessary changes, test the changes locally and post a screenshot of the result

The problem is obviously not understood by Google either -- several opened issue reports are hanging in the queues...

> 2. ensure that the main Nextcloud Android app already runs on ChromeOS

It does but I don't even know why.

I hate saying so but from the perspective of a user I can only see "some Android applications can be installed while I'm told about others that they are "not compatible with my device" and must not be installed". And in these cases the line is exactly between any other "ChromeOS device" and "ChromeBook Plus". It's affecting quite a few applications and as there is no more detailed message from the mechanism that decides it.

> 3. you are commit to maintain this feature as we have no ChromeOS available for testing purposes

This is completely irrelevant. Testing whether I can install the application will be automatic as it would fail on updates as soon as Android applications are updated. Neither can I test it. If the Play Store is not providing any additional information about users who would like to download something and *why* selling it was impossible to the seller we can just as well close the issue.

> Given the Single Sign On mechanism

As I said: I do not get that far.

**Comment by stefan-niedermann** [comment]
> The problem is obviously not understood by Google either -- several opened issue reports are hanging in the queues...

Well, I don't have any influence on opened issue reports anywhere outside of this project, obviously.
You requested to

> Add the necessary capability to your manifests that permits installation not only on ChromeOS but also on the stub-type "Chromebook Plus"

and given I have no idea how and which  capabilities need to be added, I asked you to open a Pull Request. I have no clue what do add to the Manifest and I have unfortunately no time for deep diving into this topic.

> I hate saying so but from the perspective of a user I can only see "some Android applications can be installed while I'm told about others that they are "not compatible with my device" and must not be installed". And in these cases the line is exactly between any other "ChromeOS device" and "ChromeBook Plus". It's affecting quite a few applications and as there is no more detailed message from the mechanism that decides it.

From the perspective of the developer I can say, that we do not prohibit any platforms explicitly, neither in the app nor in the Play Store configuration. Installing the [APK file directly](https://f-droid.org/repo/it.niedermann.nextcloud.deck_1023004.apk) might give us some clue whether the apps Manifest or the Play Store is the actual cause.

> This is completely irrelevant.

No it is not. I hope you are able to relate that I will not add anything to the Manifest of the app that I don't understand *and* can not test.

> Testing whether I can install the application will be automatic as it would fail on updates as soon as Android applications are updated.

So we support something until it breaks one day in the future? When we have some user base and it breaks, no one will be able to understand or fix the root cause. This is not what I would call "support".

> If the Play Store is not providing any additional information about users who would like to download something and why selling it was impossible to the seller we can just as well close the issue.

1. Not sure how to say that polite, so let me say it without any offense directly: I do not care about your three buckets. I just posted a link to the fully working `.apk` file.
2. Play Store does indeed not provide any information as far as I can see.

---

Seriously, the Deck Android doesn't work on your ChromeBook Plus device. That's a pity. But coming here basically saying

> Do something with the manifest, I have no idea what or do something on the Play Store or anything else!

won't work. I am open to close this edge case, but you will need to say me what to do here. It's you environment that has issues after all. As said before I don't have access to a ChromeBook.

**TL;DR: There will only be a solution for your issue if you *contribute* it.**

**Comment by noseshimself** [comment]
> From the perspective of the developer I can say, that we do not prohibit any platforms explicitly, neither in the app nor in the Play Store configuration.

Ok, that's at least an additional hint.

> So we support something until it breaks one day in the future?

It seems you are already doing that -- installing it was possible on any ChromeOS device until the ChromeBook Plus came along.

> When we have some user base and it breaks, no one will be able to understand or fix the root cause. This is not what I would call "support".

The root cause it that a new device class showed up that is being treated differently by Google Play and the only entities that could get this fixed are the developers putting their apps in that store and can't get them downloaded by certain devices.

So opening a ticket like "why is a ChromeOS device of type BRYA/REDRIX running brya-release/R120-15662.76.0 deemed incompatible with my application" for the Lords of the Play Store is the only way to fix it. And this is definitely nothing they are answering if I ask them.

> 1. Not sure how to say that polite, so let me say it without any offense directly: I do not care about your three buckets. I just posted a link to the fully working `.apk` file.

The link is nevertheless useless for a ChromeOS device. Side-loading anything is damaging the security concept of ChromeOS and requires opening a serious hole in system security. And even if I did it to install this apk it would not really get at the root of the problem (the seeming incompatibility with certain ChromeOS devices although all of them have the same vanilla Android VM). This has to be solved by Google and as far as I ca see opening an issue for that area is limited to application developers.
